wagtail/wagtailcore/wagtail_hooks.py

```diff
@@ -1,8 +1,17 @@
 from __future__ import absolute_import, unicode_literals
 
+from django.conf import settings
+from django.contrib.auth.views import redirect_to_login
 from django.core.urlresolvers import reverse
 
+from wagtail.utils.compat import user_is_authenticated
 from wagtail.wagtailcore import hooks
+from wagtail.wagtailcore.models import PageViewRestriction
+
+
+def require_wagtail_login(next):
```

Comment:
For a multi-domain site (which is probably the most common case where you'd want to vary the login page based on section) you could check `request.site` within the login view / template - or, more neatly, use wagtail.contrib.settings - since the redirect to the login page will keep you on the same domain. In other cases, you could (hackily) inspect the 'next' parameter.

I can't see a particularly nice way of making the URL switchable as a feature provided by Wagtail itself (where 'nice' = 'not hard-coding a list of URL paths or IDs in the settings file')...

I did consider making this configurable on a per-page-type basis, as we do for password-based restrictions - I couldn't think of a use-case for that at the time, but I guess paywalls are the big one (display the first paragraph of an article, then "log in to view the rest of this article"). I left that out in the interest of keeping this an MVP, though.